### PR TITLE
Update Tideways version and debian repo

### PR DESCRIPTION
**Pull request by edorian**
## PR Description

Hey Folks,

Volker from Tideways with a small PR to update our Repository for Debian packages.

We changed our Repo from AWS to Cloudsmith and our own domain. The old one is still available for a while so there is no rush or anything, we just wanted to reach up and get this updated :) 

You can validate that this is the new Tideways repository at https://tideways.com/profiler/downloads where it's also listed and in case of doubt feel free to reach out to support@tideways.com for validation.

Thank you!

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Review by alexanderM91 — APPROVED** [review]

**Comment by edorian** [comment]
Is there something I can do to help move this forward / get this merged?

If it's still waiting for your QA and all is good: Sorry for the ping, but thought I ask :)

**Comment by alexanderM91** [comment]
> Is there something I can do to help move this forward / get this merged?
> 
> If it's still waiting for your QA and all is good: Sorry for the ping, but thought I ask :)

Hey @edorian,
No, thanks you already did an excellent work. 

It's in our QA queue and waiting for QA. As soon as it will be finished, we are ready to go.

File: debian/bullseye/7.4/Dockerfile [hunk 1]
```diff
@@ -127,8 +127,8 @@ RUN curl -L "https://download.newrelic.com/php_agent/archive/${NEWRELIC_VERSION}
   && ln -s /lib64/ld-linux-x86-64.so.2 /lib/ld-linux-x86-64.so.2
 
 # Tideways
-RUN bash -c 'if [ "${TARGETPLATFORM}" == "linux/amd64" ]; then echo "deb http://s3-eu-west-1.amazonaws.com/tideways/packages debian main" > /etc/apt/sources.list.d/tideways.list && \
-   curl -sS "https://s3-eu-west-1.amazonaws.com/tideways/packages/EEB5E8F4.gpg" | apt-key add - && \
+RUN bash -c 'if [ "${TARGETPLATFORM}" == "linux/amd64" ]; then echo "deb https://packages.tideways.com/apt-packages-main any-version main" > /etc/apt/sources.list.d/tideways.list && \
```
Comment (alexanderM91) [inline debian/bullseye/7.4/Dockerfile:130]: Hi @edorian, 
sorry, but I have some concerns about this line.
Did you have a chance to build it?

File: debian/bullseye/8.0/Dockerfile [hunk 2]
```diff
@@ -126,8 +126,8 @@ RUN curl -L "https://download.newrelic.com/php_agent/archive/${NEWRELIC_VERSION}
   && ln -s /lib64/ld-linux-x86-64.so.2 /lib/ld-linux-x86-64.so.2
 
 # Tideways
-RUN bash -c 'if [ "${TARGETPLATFORM}" == "linux/amd64" ]; then echo "deb http://s3-eu-west-1.amazonaws.com/tideways/packages debian main" > /etc/apt/sources.list.d/tideways.list && \
-   curl -sS "https://s3-eu-west-1.amazonaws.com/tideways/packages/EEB5E8F4.gpg" | apt-key add - && \
+RUN bash -c 'if [ "${TARGETPLATFORM}" == "linux/amd64" ]; then echo "deb https://packages.tideways.com/apt-packages-main any-version main" > /etc/apt/sources.list.d/tideways.list && \
```
Comment (alexanderM91) [inline debian/bullseye/8.0/Dockerfile:129]: Hey, 
sorry, but I have some concerns about this line.
Did you build it locally?

Comment (edorian) [reply debian/bullseye/8.0/Dockerfile:129]: Hey Alexander,

What are your concerns here? Happy to answer all questions you might have.

We recommend our customers to build this in a similar way: https://support.tideways.com/documentation/setup/installation/docker-with-compose.html#tideways-daemon-dockerized

Are you concerned about the `any-version` ? We don't publish specific packages for debian/ubuntu (and never have) or for specific releases as the `deb`s work on all of these systems.

As nothing else changed here except the URLs and `debian` -> `any-version` I'm not sure what else you could be referring too; sorry If I'm missing something here!

> Did you build it locally?

Yes. I had to work around https://download.newrelic.com/php_agent/archive/9.19.0.309/newrelic-php5-9.19.0.309-linux.tar.gz not being available but I recall no other errors.

Comment (edorian) [reply debian/bullseye/8.0/Dockerfile:129]: I did another run and I see what you're referring too now.

The key adding fails as curl doesn't follow the, new, redirect here. Sorry for not thinking of that when we created a redirect and thank you very much for catching it!

**Comment by alexanderM91** [comment]
Hey @edorian, small updates from our side.
Unfortunately, during the QA we faced some error just for PHP 8.1.
We are trying to figure out the root cause. 

```PHP Warning:  PHP Startup: Unable to load dynamic library 'tideways.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20210902/tideways.so (/usr/local/lib/php/extensions/no-debug-non-zts-20210902/tideways.so: cannot open shared object file: No such file or directory), /usr/local/lib/php/extensions/no-debug-non-zts-20210902/tideways.so.so (/usr/local/lib/php/extensions/no-debug-non-zts-20210902/tideways.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0```

We are on the investigation stage.

**Comment by edorian** [comment]
Thanks for letting me know @alexanderM91 

Running `docker build -t baz -f debian/bullseye/8.1/Dockerfile .` seems to work for me

```
 ✓ $ docker run --rm -it baz php -v
PHP 8.1.8 (cli) (built: Aug  2 2022 07:20:57) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.8, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.8, Copyright (c), by Zend Technologies

 ✓ $ docker run --rm -it baz php -m | grep tideways
tideways
```

Also testing the php-fpm 

```
root@92ba8cc7d659:/data# SCRIPT_NAME=/fpm-ping.php SCRIPT_FILENAME=/fpm-ping.php REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000
Content-type: text/plain;charset=UTF-8
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Cache-Control: no-cache, no-store, must-revalidate, max-age=0

OK
```

So did 

```
echo "<?php phpinfo(); " > /var/www/html/index.php

SCRIPT_NAME=/ SCRIPT_FILENAME=/var/www/html/index.php DOCUMENT_ROOT=/var/www/html/ REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000
```
which is also loading the config from `/usr/local/etc/php/conf.d/tideways.ini`. 

Sorry I can't be more helpful here but from what I can test things seem to be working.

The extension dir PHP-FPM uses seems to be set correct and the tideways file is correctly linked here by our installer:
```
root@92ba8cc7d659:/data# SCRIPT_NAME=/ SCRIPT_FILENAME=/var/www/html/index.php DOCUMENT_ROOT=/var/www/html/ REQUEST_METHOD=GET cgi-fcgi -bind -connect 127.0.0.1:9000 | grep extension
<tr><td class="e">extension_dir</td><td class="v">/usr/local/lib/php/extensions/no-debug-non-zts-20210902</td><td class="v">/usr/local/lib/php/extensions/no-debug-non-zts-20210902</td></tr>

root@92ba8cc7d659:/data# ls -lash /usr/local/lib/php/extensions/no-debug-non-zts-20210902/tideways.so 
0 lrwxrwxrwx 1 root root 37 Oct 24 16:50 /usr/local/lib/php/extensions/no-debug-non-zts-20210902/tideways.so -> /usr/lib/tideways/tideways-php-8.1.so
```

If you think I could help feel free to point me to a container/reproduce. Otherwise no worries; there is no rush from our side :)

**Comment by alexanderM91** [comment]
Hi @edorian, 

with the current image we came across with another issue - Tideways extension cannot be loaded properly. 
Screenshot 2022-10-26 at 09 51 31

We did some adjustments for the current PR. You can find them in the attached one.
Thank you.

**Comment by edorian** [comment]
Curious, as I couldn't reproduce the issue with the Dockerfiles in this branch. 

While this is a non recommended install method for Debian systems it also works and when using direct downloads the URLs are also fine :) 

So I guess you can close this then :) 

I'd be happy to look into the installation issue and fix it if you could provide me with reproduce commands. But that's fully optional of course.

**Comment by alexanderM91** [comment]
@kalaputsik, could you please provide command which will help to reproduce a command?
Thank you

**Comment by alexanderM91** [comment]
Hi @edorian, 

you can test your image via command

```
$ docker run -i --rm {image_name} bash -s<<'EOF'
    docker-php-source extract
    echo "Installed extensions";
    echo "====================";
    for ext in `ls /usr/src/php/ext`; do echo ' ' `php -r "if (extension_loaded('$ext' !== 'opcache' ? '$ext' : 'Zend OPcache')) { echo '[x] $ext'; } else { echo '[ ] $ext'; }"`; done
    echo "";
    echo "Disabled extensions";
    echo "====================";
    for f in /usr/local/etc/php/disabled/*.ini; do disabled=$(basename $f | sed -e 's/\.ini$//'); echo " [ ] ${disabled} $(PHP_INI_SCAN_DIR=:/usr/local/etc/php/disabled php -r "echo phpversion('${disabled}');")"; done
    echo "";
    pear list -c pecl
    echo "";
    echo "Composer";
    echo "====================";
    composer -V
EOF```

**Comment by edorian** [comment]
I see, thank you for the reproduce. That makes sense. The tideways installer puts the .ini file in a folder that is scanned and then another folder with the same file is added.

`php --ini` in that moment shows me both
```
/usr/local/etc/php/conf.d/tideways.ini,
/usr/local/etc/php/disabled/tideways.ini
```
being loaded.

Thank you for letting me know. It's good to confirm that is not an issue with our installer :) 

As you are setting this up another way i'd say it makes sense to close this PR then :)

**Comment by alexanderM91** [comment]
Hey, close the PR as the changes already in master branch.
Thanks a lot for your contribution and problems solving support.
Have a great evening.

**Pull request closed without merging**